### ISSUE TITLE
Changing Parent Value Of Child Slots Makes Fire On Change Fail.

### ISSUE THREAD

**Issue by AmasterAmaster** [Yellow-Dog-Man/Resonite-Issues#2834]
### Describe the bug?

When changing the child slot reference inputs for the `Get Parent Slot` node while also connected to a `Fire On Change` node, there will not be an impulse.

This happens every time when making a fresh new world (tested on a grid world), setting it up exactly like this will make it fail. I have tested it with the following nodes and methods to see if it would make it work:
- Continuously Changing Relay (failed - no difference)
- Tested with the parent field (as a source node) while moving the child slot to a new parent (this works fine as it is direct)
- Moved the child slot from one parent to another (this works fine)
- Multiplex (This made it work and removes the bug for this instance of the world, i don't know why)

Another strange thing about this: After connecting the multiplex to the node group (before the `Get Parent Slot` node), it works, then I removed that node, and it continued to work. So there is something with multiplex that makes these set of nodes respond better. I don't know what is happening under the hood with that, but that can be a lead.

### To Reproduce

- Spawn two empty slots then make a child slot in both of them
- Create the setup shown in the screenshots section (two child slot references -> `Get Parent Slot` -> `Fire On Change` -> `Pulse Display`)
- Attempt to swap the input wires for both child slot references

Additional steps to temporary remove the bug in the world:
- Spawn a `Multiplex` node for slots
- Attach it before the `Get Parent Slot` node
- Connect the references to the multiplex along with setting up the value input node (the number to swap the references)
- (After switching numbers around) remove the multiplex and connect on of the child references like before, then swap the input of the other reference manually, you may notice that the pulse display is working now, even though there was no change between before and after the `Multiplex` steps.

### Expected behavior

Switching child slot references should make this `Fire On Change` node pulse when detecting a different parent.

### Screenshots

![2024-08-29 15 30 36](https://github.com/user-attachments/assets/4f9348b4-c77e-4da4-a83d-4bd4904f09b1)

### Resonite Version Number

Beta 2024.8.5.1341

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Index & Desktop

### Log Files

[ANDREW-SUPERPC - 2024.8.5.1341 - 2024-08-29 10_55_12.log](https://github.com/user-attachments/files/16801910/ANDREW-SUPERPC.-.2024.8.5.1341.-.2024-08-29.10_55_12.log)

### Reporters

AmasterAmaster

**Comment by shiftyscales** [comment]
> Switching child slot references should make this Fire On Change node pulse when detecting a different parent.

Based on your report above- it sounds like it does work correctly though as long as all of the nodes remain connected? Why are you concerned with the use-case of it firing when the user attaches a new input to it directly while building it?

Using a multiplexer, or a store would be the expected ways I'd anticipate a user using these nodes if you need to have multiple input sources that it is detecting.

Does the node work as intended in the setup you provided if you move the slot between parents by manually reparenting it rather than changing what slot's parent is being read?

**Comment by AmasterAmaster** [comment]
> it sounds like it does work correctly though as long as all of the nodes remain connected?

Yes @shiftyscales, as this issue is specifically talking about the connections and interaction between the nodes. And then it falls apart when trying to change the slot reference directly by connecting a new slot wire (and since it is a different slot under a different parent, I would expect it to register it and pulse accordingly).

> Why are you concerned with the use-case of it firing when the user attaches a new input to it directly while building it?

Just for the reason of testing a group of nodes with different slots, and wanting to see what would happen as well as what results I would get from it. Since I discovered this oddity, I might as well report it, you know? A user would see that a parent has changed and be confused on why it would not fire when providing a different parent.

> Does the node work as intended in the setup you provided if you move the slot between parents by manually reparenting it rather than changing what slot's parent is being read?

Yes, and I suspect it is because the parent field in the slot is changing and the `Fire On Change` node is registering that fine. While switching the slots instead by wire, there could be a gap of time preventing the node from firing, but as seen with the multiplex, that fixes it (only compounding the confusion when removing that node with the exact same setup, and still working as if the bug was not there).

It is a weird phenomenon.
